src/server/game/Entities/AreaTrigger/AreaTrigger.h

```diff
@@ -37,6 +40,12 @@ class TC_GAME_API AreaTrigger : public WorldObject, public GridObject<AreaTrigge
         void AddToWorld() override;
         void RemoveFromWorld() override;
 
+        bool AIM_Initialize(AreaTriggerAI* ai = nullptr);
```

Comment:
complicates the code a bit - by not exposing such functionality (no SetAI) you dont also need to care about _AI_locked (maybe you don't even need IsAIEnabled, just check for null)
also note for naming
AIM (M is movement, creature calls Motion_Initialize there)

actually you dont need 80% of that creature copypasta, method arg is unneeded, AIM_Destroy is unneeded